src/backend/utils/adt/float.c

```diff
@@ -497,32 +496,9 @@ float8in(PG_FUNCTION_ARGS)
 			 errmsg("invalid input syntax for type double precision: \"%s\"",
 					orig_num)));
 
-	/*
-	 * strtod does not support values from 1e-323 to 1e-308 for double datatype in rhel6
-	 * due to changes in glibc. In order to overcome this issue we use strtold to parse
-	 * the string and store it in long double so that we get higher precision. We check if
-	 * this value is within allowed range for double using the below two checks:- 
-	 *
-	 * 1) Overflow - When we try to store a value > 1e308 in a double datatype, it gets represented 
-	 * internally as "inf". Hence in order to check for overflow, we use the isinf() method in order
-	 * to check if (double) val has overflowed. But "inf" is a legitimate value for float8, hence if the
-	 * user entered "inf"/"Infinity" we allow it. Otherwise we treat it as overflow.
-	 *
-	 * 2) Underflow - When we try to store a value < 1e-323, it gets converted to a 0.0. But "0.0" is a 
-	 * legitimate value allowed by float8 datatype. Hence we examine if the val is > 0 or < 0 with higher 
-	 * precision (as long double). Again, 0.0 is allowed for float8. If the user entered 0.0, val will be 0.0
-	 * in both higher precision and when rounded down and we allow it. Otherwise in higher precision, 
-	 * val will be > 0 or < 0 and when rounded down to double it will be 0 which is treated as underflow.
-	 */
-
-	if ( pg_strncasecmp(num, "Infinity", 8) != 0  && pg_strncasecmp(num, "-Infinity", 9) != 0 &&\
-		 pg_strncasecmp(num, "Inf", 3) != 0 && pg_strncasecmp(num, "-Inf", 4) != 0 &&\
-		 pg_strncasecmp(num, "+Infinity", 9) != 0 && pg_strncasecmp(num, "+Inf", 4) != 0 )
-		literal_inf = false;
-
-	CHECKFLOATVAL((double) val, literal_inf, !(val > 0 || val < 0));
+	CHECKFLOATVAL(val, true, true);
```

Comment:
IMO this check with the second and third `true` arguments are redundant here. Most likely, it corresponded to removed block of check that product from `strtold()` more than 1e308 and less than 1e-323

Reply:
In vanilla postgres this check [was deleted](https://github.com/postgres/postgres/commit/50861cd683e86d5ef2dc1cb669fb503225e4eb98#diff-6356c43bf114bc5b369668aa59b5f71fdc2691508d30c9b207d8cfd106164ae4L537) as part of global refactoring of float module

Reply:
@maksm90 - I have made a small git blame research in PG code and found out that this check appeared in [8.3](https://github.com/postgres/postgres/commit/f9ac414c35e). It seems to be a wrong replacement of the old check: previously we checked that the non-infifnty float value is not out of range. As you have mentioned, the current check makes no sense - it was removed in [9.5](https://github.com/postgres/postgres/commit/50861cd683e86d5ef2dc1cb669fb503225e4eb98). So, let's get rid of it.